aten/src/ATen/native/xnnpack/Allocator.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <ATen/native/xnnpack/Common.h>
+
+#ifdef USE_XNNPACK
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+
+// XNNPACK can out-of-bound access XNN_EXTRA_BYTES beyond the end of the input
+// and output tensors.  This behavior will trigger ASAN, and can result in a
+// segfault if the accessed memory just so happens to fall on a different page
+// the current process has no read access to.
+// Here we define a custom allocator that allocates the extra storage required
+// to keep this behavior safe.
+
+class Allocator final : public at::Allocator {
+ public:
+  Allocator() = default;
+  virtual ~Allocator() override = default;
+
+  static void deleter(void* buffer);
+
+  virtual DataPtr allocate(size_t nbytes) const override;
+  virtual DeleterFnPtr raw_deleter() const override;
+
+ private:
+  static constexpr uint32_t kGuard = XNN_EXTRA_BYTES;
+};
+
+at::Tensor empty(
```

Comment:
it's pretty confusing when reading the code to disambiguate at::empty and at::native:xnnpack::internal::empty. Can you name this function differently (empty_with_tail_padding)?